apc_sma.c

```diff
@@ -59,6 +59,14 @@ struct sma_header_t {
 #define SMA_RO(sma, i)   ((char*)(sma->segs[i]).roaddr)
 #define SMA_LCK(sma, i)  ((SMA_HDR(sma, i))->sma_lock)
 
+#define SMA_CREATE_LOCK  APC_CREATE_MUTEX
+#define SMA_DESTROY_LOCK APC_DESTROY_MUTEX
+#define SMA_LOCK(sma, i) (APC_MUTEX_LOCK(&SMA_LCK(sma, i)))
+#define SMA_UNLOCK(sma, i) (APC_MUTEX_UNLOCK(&SMA_LCK(sma, i)))
+
+#define SMA_RLOCK(sma, i) (APC_MUTEX_RLOCK(&SMA_LCK(sma, i)))
+#define SMA_RUNLOCK(sma, i) (APC_MUTEX_RUNLOCK(&SMA_LCK(sma, i)))
```

Comment:
I don't think keeping the read-lock distinction here is worthwhile. We should not expose read-lock ability in the mutex API (even if it is ultimately backed by an rwlock).